cpp/src/arrow/adapters/orc/adapter.cc

```diff
@@ -530,7 +530,7 @@ ORCFileReader::~ORCFileReader() {}
 
 Result<std::unique_ptr<ORCFileReader>> ORCFileReader::Open(
     const std::shared_ptr<io::RandomAccessFile>& file, MemoryPool* pool) {
-  auto result = std::unique_ptr<ORCFileReader>(new ORCFileReader());
+  auto result = std::unique_ptr<ORCFileReader>(new ORCFileReader);
```

Comment:
Can we use `make_unique` here?

Reply:
Can't because ORCFileReader's ctor is private. This delete of () was by mistake and I reverted it.

Reply:
I see.